galaxy/main/models/collection.py

```diff
@@ -59,7 +59,7 @@ class Collection(mixins.TimestampsMixin, models.Model):
     # References
     latest_version = models.ForeignKey(
         'CollectionVersion',
-        on_delete=models.PROTECT,
+        on_delete=models.SET_NULL,
```

Comment:
This affects delete of the `CollectionVersion`, but you don't add that as a registered model for the admin in this PR.

Reply:
There is circular dependency between Collection and CollectionVersion.
You cannot delete `Collection` if it has `latest_version` field value other than null.